ui/dashboards/src/components/Dashboard.tsx

```diff
@@ -26,14 +27,19 @@ export interface DashboardProps extends BoxProps {
 export function Dashboard(props: DashboardProps) {
   const { spec, ...others } = props;
 
+  const { layouts } = useLayouts();
+  const { getDashboard } = useDashboard();
+
   return (
     <Box {...others}>
       <ErrorBoundary FallbackComponent={ErrorAlert}>
-        {spec.layouts.map((layout, idx) => (
+        {layouts.map((layout, idx) => (
           <GridLayout
             key={idx}
             definition={layout}
-            renderGridItemContent={(definition) => <GridItemContent content={definition.content} spec={spec} />}
+            renderGridItemContent={(definition) => (
+              <GridItemContent content={definition.content} spec={getDashboard() || spec} />
```

Comment:
Not sure I understand this? Is `getDashboard` just for edit mode while `spec` is for view mode? Feels like maybe this component should just render whatever is passed in via props? If I were using the component (passing in a spec) but then it stopped rendering my spec, I think I'd be surprised by that behavior?

Reply:
you're right, i reverted the change and had to move the components out of <ViewDashboard /> component into <DashboardApp /> so that I could call `useDashboard` inside the `DashboardStoreProvider` and pass the dashboard spec into this Dashboard component